File: codewars/python/dirReduc.py
def dirReduc(u):
	op = {"SOUTH": "NORTH", "WEST":"EAST", "EAST":"WEST", "NORTH":"SOUTH"}

	for i in u:
		n = u.index(i)
		if u[n+1] == op[i]:
			u.remove(u[n])
			u.remove(u[n+1])
		else:
			continue
	return u

a = ["NORTH", "SOUTH", "SOUTH", "EAST", "WEST", "NORTH", "WEST"]
print dirReduc(a)

File: codewars/ruby/dirReduc.rb

def dirReduc(u)
	op = {"SOUTH" => "NORTH", "WEST" => "EAST", "EAST" => "WEST", "NORTH" => "SOUTH"}
	u.each do |w|
		if op[u[u.index(w)+1]] == w
			u.delete(u[u.index(w)+1]); u.delete(w)
		end 
	end
	u
end

a = ["NORTH", "SOUTH", "SOUTH", "EAST", "WEST", "NORTH", "WEST"]
puts dirReduc(a)

